### PR TITLE
Add Block#isLoaded method

### PR DESCRIPTION
**Pull request by sopel39**
Not only LazyBlocks can be loaded/not loaded.
Some blocks (e.g. dictionary blocks) can be
partially loaded.

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/operator/project/MergePages.java [hunk 1]
```diff
@@ -172,17 +171,17 @@ private Page flush()
             return output;
         }
 
-        private static boolean containsNotLoadedLazyBlock(Page page)
+        private static boolean allBlocksLoaded(Page page)
```
Comment (Praveen2112) [inline presto-main/src/main/java/io/prestosql/operator/project/MergePages.java]: Can we move this logic to `Page` ? As other operators can use the same in the future

Comment (sopel39) [reply presto-main/src/main/java/io/prestosql/operator/project/MergePages.java]: This method is part of rather poor heuristics and not used anywhere else. Let's see it if becomes useful in other places.

File: presto-main/src/main/java/io/prestosql/operator/PageUtils.java [hunk 2]
```diff
@@ -30,37 +30,30 @@ public static Page recordMaterializedBytes(Page page, LongConsumer sizeInBytesCo
...
         Block[] blocks = new Block[page.getChannelCount()];
         long loadedBlocksSizeInBytes = 0;
-        boolean allBlocksNonLazy = true;
+        boolean allBlocksLoaded = true;
 
         for (int i = 0; i < page.getChannelCount(); ++i) {
             Block block = page.getBlock(i);
-            if (block instanceof LazyBlock) {
-                LazyBlock delegateLazyBlock = (LazyBlock) block;
-                if (delegateLazyBlock.isLoaded()) {
```
Comment (dain) [inline presto-main/src/main/java/io/prestosql/operator/PageUtils.java]: Looks like we lost this case, which seems fine to me.  If this is a behavior change maybe move it to a separate commit, got keep this one more mechanical.
